code/game/objects/structures/crates_lockers/closets/cardboardbox.dm

```diff
@@ -18,12 +18,14 @@
 /obj/structure/closet/cardboard/relaymove(mob/user, direction)
 	if(opened || move_delay || user.stat || user.stunned || user.weakened || user.paralysis || !isturf(loc) || !has_gravity(loc))
 		return
-	move_delay = 1
-	if(step(src, direction))
+	move_delay = TRUE
+	var/oldloc = loc
+	step(src, direction)
```

Comment:
did you try what TG did with `if(step(src, direction))` then using timers?

Reply:
the issue isn't timers, the if check is the reason you have no move delay when you move diagonally against walls

Reply:
I'm...not suggesting the issue is with timers.

Reply:
Please make this:

```
/obj/structure/closet/cardboard/relaymove(mob/living/user, direction)
	if(!istype(user) || opened || move_delay || user.incapacitated() || !isturf(loc) || !has_gravity(loc))
		return
	move_delay = TRUE
	var/oldloc = loc
	step(src, direction)
	if(oldloc != loc)
		addtimer(CALLBACK(src, .proc/ResetMoveDelay), config.walk_speed)
	else
		move_delay = FALSE

/obj/structure/closet/cardboard/proc/ResetMoveDelay()
	move_delay = FALSE
```